### PR TITLE
[docs] Publish a move module using ts sdk in my first dapp

### PR DESCRIPTION
**Pull request by 0xmaayan**
### Description
Adds instruction on how to publish a package using the TS SDK in `my first dapp` guide

### CODE REVIEW

File: developer-docs-site/docs/tutorials/first-dapp.md [hunk 1]
```diff
@@ -368,6 +368,217 @@ You can also verify that the module was published by going to the [Aptos Explore
...
+cat package-metadata.bcs | od -v -t x1 -A n | tr -d ' \n'
+```
+That will output a hex string we can later use.
+
+Convert `message.mv` to a hex string. On Mac and Linux we can use the command
+```bash
+cat bytecode_modules/message.mv | od -v -t x1 -A n | tr -d ' \n'
+```
+That will output a hex string we can later use.
+
```
Comment (gregnazario) [inline developer-docs-site/docs/tutorials/first-dapp.md:428]: Should we just load these directly from the files into Typescript?  Might be easier than first converting to Hex, and then using them.

File: developer-docs-site/docs/tutorials/first-dapp.md [hunk 2]
```diff
@@ -368,6 +368,217 @@ You can also verify that the module was published by going to the [Aptos Explore
...
+      // ...
+      <div>
+        <button onClick={onPublishModule} disabled={isPublishing}>
+          Publish Package
+        </button>
+        {publishPackageTxnHash && (
+          <div>
+            <p>
+              <a
+                href={`https://fullnode.devnet.aptoslabs.com/v1/accounts/${address}/modules`}
```
Comment (gregnazario) [inline developer-docs-site/docs/tutorials/first-dapp.md:494]: Other thing is we probably need to check if this is all done on devnet / ensure it's consistent.  Last time I looked I think it was devnet, but just to double check

File: developer-docs-site/docs/tutorials/first-dapp.md [hunk 3]
```diff
@@ -368,6 +368,217 @@ You can also verify that the module was published by going to the [Aptos Explore
...
+            // modules
+            "<message.mv hex string>"
+          ).toUint8Array()
+        ),
+      ]
+    );
+
+    await client.waitForTransaction(txnHash);
+    setIsPublishing(false);
+    setPublishPackageTxnHash(txnHash);
```
Comment (gregnazario) [inline developer-docs-site/docs/tutorials/first-dapp.md:480]: Let's put comments to tell what these last couple lines mean

File: developer-docs-site/docs/tutorials/first-dapp.md (hunk 1 above)
Comment (0xmaayan) [reply developer-docs-site/docs/tutorials/first-dapp.md:428]: yes we cam, same as I started to do in the ANS implementation https://github.com/aptos-labs/aptos-core/pull/6000/files#diff-af27ba8aea8ccbfc871dd0e005811cc7198ad93d52686f017e79bc15483c96bdR46-R58 
we still need to compile them first though - can use `aptos move compile` for that

File: developer-docs-site/docs/tutorials/first-dapp.md (hunk 2 above)
Comment (0xmaayan) [reply developer-docs-site/docs/tutorials/first-dapp.md:494]: Agreed, i think it does use devnet. maybe we should decide to use devnet only in this tutorial and point links/explorer to devnet

File: developer-docs-site/docs/tutorials/first-dapp.md (hunk 1 above)
Comment (davidiw) [reply developer-docs-site/docs/tutorials/first-dapp.md:428]: On a side note... I think there's a project that would make the SDKs work more seamlessly with the CLI
* Make local testing faster
* Add support for compilation and publishing
* Not need this interesting thing

Comment (0xmaayan) [reply developer-docs-site/docs/tutorials/first-dapp.md:428]: The TS SDK? Interesting..

**Review by clay-aptos — APPROVED** [review]
Thank you so much for making these improvements, Maayan! This LGTM with some edits I made directly. Sorry for the delay. Thanks again!

**Review by gregnazario — APPROVED** [review]
Let's roll with this for now, but we should work on tooling to make the hex inputs easier to get.

**Comment by clay-aptos** [comment]
> Thank you so much for making these improvements, Maayan! This LGTM with some edits I made directly. Sorry for the delay. Thanks again!

Well, I guess I didn't submit my edits. See the new PR:
https://github.com/aptos-labs/aptos-core/pull/6164
Thanks!

**Pull request merged**